tests/unit/plugins/kubernetes/kubernetes_task_config_test.py

```diff
@@ -0,0 +1,22 @@
+from pyrsistent import InvariantException
+
+from task_processing.plugins.kubernetes.task_config import KubernetesTaskConfig
+
+
+def test_kubernetes_task_config_set_pod_name():
+    k1 = KubernetesTaskConfig(name="fake_pod_name")
+    k2 = KubernetesTaskConfig(name="fake_pod_name2")
+
+    try:
+        k1 = k1.set(name='a'*254)
+        assert False, 'Pod name must have up to 253 characters.'
+    except InvariantException as e:
+        print(e)
+        assert True
+
+    try:
+        k2 = k2.set(name="INVALID"+k2.name)
+        assert False, 'Must comply with Kubernetes pod naming standards.'
+    except InvariantException as e:
+        print(e)
+        assert True
```

Comment:
++ on writing tests!

suggestion: we probably want to split this into two tests just so that it's easier to see exactly what's broken should we make a bad refactor somewhere at some point. additionally, pytest has a nice wrapper for testing exceptions (https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions) that we might want to use. maybe something like:

```suggestion
def test_kubernetes_task_config_set_pod_name_rejects_long_name():
    task_config = KubernetesTaskConfig(name="fake_pod_name")

    with pytest.raises(InvariantException):
        task_config.set(name='a' * 254)

def test_kubernetes_task_config_enforces_kubernetes_name_requirements(     
    task_config = KubernetesTaskConfig(name="fake_pod_name")

    with pytest.raises(InvariantException):
        k2.set(name=f"INVALID{task_config.name})
```

Reply:
(we'll need to add an `import pytest` line to the top of this file, of course)

Reply:
Done.